modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -112,6 +112,12 @@ CV__DNN_INLINE_NS_BEGIN
      */
     CV_EXPORTS void enableModelDiagnostics(bool isDiagnosticsMode);
 
+    /**
+     * @brief Skip model import after diagnostic run in readNet() functions.
+     * @param[in] skip Indicates whether to skip the import.
+     */
+    CV_EXPORTS void skipModelImport(bool skip);
```

Comment:
Why do we need that as part of public API?

Reply:
This function is used in `test_tf_importer.cpp` and prevents tests of diagnostic mode from failing. I agree that users probably shouldn't care about this. I couldn't move the definition to `dnn_common.hpp` because of undefined reference error(implementation was placed in `dnn.cpp`). Can you suggest a better place for it or how to get around this problem?

As for the contents of `layer_reg.private.hpp`, I think removing it would be a breaking change, and so if someone is using it, they should be able to use it thread-safely.